conductr_cli/bndl_utils.py

```diff
@@ -75,74 +74,154 @@ def try_match(pattern, slice):
 
-    if args.tag not in tags and args.tag not in BNDL_IGNORE_TAGS:
-        tags.append(args.tag)
+    if args_compatibility_version is not None:
+        config.put('compatibilityVersion', args_compatibility_version)
+    if with_defaults and 'compatibilityVersion' not in config:
+        config.put('compatibilityVersion', BNDL_DEFAULT_COMPATIBILITY_VERSION)
 
-    config.put('tags', tags)
+    if args_disk_space is not None:
+        config.put('diskSpace', args_disk_space)
+    if with_defaults and 'diskSpace' not in config:
+        config.put('diskSpace', BNDL_DEFAULT_DISK_SPACE)
 
-    annotations_tree = ConfigTree()
+    if args_memory is not None:
+        config.put('memory', args_memory)
+    if with_defaults and 'memory' not in config:
+        config.put('memory', BNDL_DEFAULT_MEMORY)
 
-    for annotation in args.annotations:
-        if '=' not in annotation:
-            raise ValueError('Invalid annotation format. Specify as name=value')
+    if args_name is not None:
+        config.put('name', args_name)
+    if with_defaults and 'name' not in config:
+        config.put('name', BNDL_DEFAULT_NAME)
 
-        annotation_parts = annotation.split('=', 1)
-        annotations_tree.put(annotation_parts[0], annotation_parts[1])
+    if args_nr_of_cpus is not None:
+        config.put('nrOfCpus', args_nr_of_cpus)
+    if with_defaults and 'nrOfCpus' not in config:
+        config.put('nrOfCpus', BNDL_DEFAULT_NR_OF_CPUS)
 
-    config.put('annotations', annotations_tree)
+    if args_roles is not None:
+        config.put('roles', args_roles)
+    if with_defaults and 'roles' not in config:
+        config.put('roles', BNDL_DEFAULT_ROLES.copy())
+
+    if args_system is not None:
+        config.put('system', args_system)
+    if with_defaults and 'system' not in config:
+        config.put('system', config.get('name') if 'name' in config else BNDL_DEFAULT_NAME)
+
+    if args_system_version is not None:
+        config.put('systemVersion', args_system_version)
+    if with_defaults and 'systemVersion' not in config:
+        config.put('systemVersion', config.get('version') if 'version' in config else BNDL_DEFAULT_VERSION)
+
+    if hasattr(args, 'image_tag'):
+        tags = config.get('tags') if 'tags' in config else []
+
+        if args.image_tag is not None and args.image_tag not in tags and args.image_tag not in BNDL_IGNORE_TAGS:
+            tags.append(args.image_tag)
+            config.put('tags', tags)
+    if hasattr(args, 'tags') and len(args.tags) > 0:
+        tags = []
+
+        for tag in args.tags:
+            if tag not in tags:
+                tags.append(tag)
+
+        config.put('tags', tags)
+    if with_defaults and 'tags' not in config:
+        config.put('tags', BNDL_DEFAULT_TAGS.copy())
+
+    if args_version is not None:
+        config.put('version', args_version)
+    if with_defaults and 'version' not in config:
+        config.put('version', BNDL_DEFAULT_VERSION)
```

Comment:
Refactored to build the config in a alphabetically sorted manner. Now explicitly lists out each of the attributes instead of iterating through a map of them. This makes it easier to customize the behavior for a given attribute (such as tags) with the trade-off of it being more verbose.